vars/storybooks.groovy

```diff
@@ -0,0 +1,83 @@
+def getStorybooksBucket() {
+  return "ci-artifacts.kibana.dev/storybooks"
+}
+
+def getDestinationDir() {
+  return env.ghprbPullId ? "pr-${env.ghprbPullId}" : buildState.get('checkoutInfo').branch.replace("/", "__")
+}
+
+def getUrl() {
+  return "https://${getStorybooksBucket()}/${getDestinationDir()}"
+}
+
+def getUrlLatest() {
+  return "${getUrl()}/latest"
+}
+
+def getUrlForCommit() {
+  return "${getUrl()}/${buildState.get('checkoutInfo').commit}"
+}
+
+def upload() {
+  dir("built_assets/storybook") {
+    sh "mv ci_composite composite"
+
+    def storybooks = sh(
+      script: 'ls -1d */',
+      returnStdout: true
+    ).trim()
+      .split('\n')
+      .collect { it.replace('/', '') }
+      .findAll { it != 'composite' }
+
+    def listHtml = storybooks.collect { """<li><a href="${getUrlForCommit()}/${it}">${it}</a></li>""" }.join("\n")
+
+    def html = """
+      <html>
+        <body>
+          <h1>Storybooks</h1>
+          <p><a href="${getUrlForCommit()}/composite">Composite Storybook</a></p>
+          <h2>All</h2>
+          <ul>
+            ${listHtml}
```

Comment:
Do we need this page? What is the composite storybook missing that this gives us?

Reply:
I didn't want the links to the individual storybooks to be effectively hidden, so I built this index page to put in front of all of them. This seems particularly useful if you're only concerned about a particular storybook, because the composite takes a long time to load.